packages/core/src/modules/credentials/repository/CredentialRecord.ts

```diff
@@ -91,6 +100,7 @@ export class CredentialRecord extends BaseRecord<DefaultCredentialTags, CustomCr
       this.credentialAttributes = props.credentialAttributes
       this.autoAcceptCredential = props.autoAcceptCredential
       this.linkedAttachments = props.linkedAttachments
+      this.revocationNotification = props.revocationNotification
```

Comment:
Maybe we can make this object a bit more generic and not specifically about the revocation notification. In the future we could then just store revocation data in the object. E.g we fetch the revocation data ourselves from the ledger and store the result in the credentialrecord.

For the framework user this gives an easy place to find the revocation state whether it was fetched ourselves or informed by the issuer.

```json
{
    "date": "the date",
    "method": "notification" // or "fetched" or something (maybe "ledger", but that wouldn't work for revocation methods that are not anchored on a ledger),
    "comment": "comment only present for notification method"
}
```

What do you think?

Reply:
Hmm looking at practical use cases what is often used for non-revocation intervals? Isn't the most often used question now:now? I think according to RFC 0441 we can only ask for the `from` being the same as `to` (which gives us an exact timestamp) or we get a single to which means basically the same as an exact timestamp right?

So we would know for sure that at a specific time the credential was revoked/not-revoked. I agree it may not be 100% applicable to every use case, but I think it could be really useful to show the revocation status of a credential next to it, and e.g. maybe showing a refresh button which will fetch the latest revocation state.

Reply:
You're correct in that what we're supporting right now based off of RFC 0441 is only supplying the same from and to, such as: now:now, 30 seconds ago/30 second ago, etc. 

Right now the ability to query the ledger for a rev reg delta (to get revocation status) is exposed on the LedgerModule, which would enable what you're describing, although it's maybe not as obvious to use for this case.